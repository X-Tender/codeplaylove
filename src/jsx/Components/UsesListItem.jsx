import React from 'react';
import PropTypes from 'prop-types';
import A from 'Semantics/A';

const propTypes = {
	children: PropTypes.string.isRequired,
	link: PropTypes.string,
};

const defaultProps = {
	link: null,
};

const UsesListItem = ({ link, children }) => (
	<li className="uses__list-item">
		{link ? (
			<A href={link} target="_blank">
				{children}
			</A>
		) : (
			children
		)}
	</li>
);

UsesListItem.propTypes = propTypes;
UsesListItem.defaultProps = defaultProps;

export default UsesListItem;
