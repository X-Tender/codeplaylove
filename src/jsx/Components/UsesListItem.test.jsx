import React from 'react';
import { shallow } from 'enzyme';
import UsesListItem from './UsesListItem';

describe('UsesListItem', () => {
	const children = 'Macbook Pro';
	const link = 'https://apple.com';

	describe('With link', () => {
		const compWithLink = shallow(<UsesListItem link={link}>{children}</UsesListItem>);

		it('renders correctly', () => {
			expect(compWithLink).toMatchSnapshot();
		});

		it('renders props correctly', () => {
			expect(compWithLink.find('A').prop('href')).toEqual(link);
		});
	});

	describe('Without link', () => {
		const compWithoutLink = shallow(<UsesListItem>{children}</UsesListItem>);

		it('renders correctly', () => {
			expect(compWithoutLink).toMatchSnapshot();
		});

		it('renders props correctly', () => {
			expect(compWithoutLink.text()).toEqual(children);
		});
	});
});
