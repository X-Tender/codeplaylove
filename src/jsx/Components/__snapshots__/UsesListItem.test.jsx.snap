// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsesListItem With link renders correctly 1`] = `
<li
  className="uses__list-item"
>
  <A
    href="https://apple.com"
    target="_blank"
    title={null}
  >
    Macbook Pro
  </A>
</li>
`;

exports[`UsesListItem Without link renders correctly 1`] = `
<li
  className="uses__list-item"
>
  Macbook Pro
</li>
`;
